src/executables/sysrepo-plugind.c

```diff
@@ -567,6 +602,30 @@ main(int argc, char **argv)
     sd_notify(0, "READY=1");
 #endif
 
+    /* update pid file (if daemon) */
+    if (!debug && pidfile) {
+        int pid_len;
+        if (ftruncate(pidfd, 0)) {
+            error_print(0, "Failed to truncate PID file (%s).", strerror(errno));
+            goto cleanup;
+        }
+
+        if (asprintf(&pid, "%d\n", getpid()) < 0) {
```

Comment:
Yeah, I read that man page too. It's probably very vague because it isn't part of any standard. I couldn't imagine that it would leave a dangling pointer in `*strp`, that would just be nuts!

Reply:
I dunno about the ANSI C file streams. The trade off is more code. The `0640` permission flags in the call to `open` has to be set in yet another function call `fchmod`. It and and `lockf` requires the use of `fileno` in order to get to the file descriptor. `fchmod` also requires error checking. I'm not sure all that extra code adds any value because of the `asprintf` which likely isn't going to fail ever.

Reply:
(and when it does fail, it really shouldn't return a dangling pointer in `*strp`, and I think that would've been called upon by now!)